### ISSUE TITLE
Leaves are being loaded from schematics as decayable=true

### ISSUE THREAD

**Issue by TheGurw** [Lunatrius/Schematica#215]
TL;DR: If your leaf blocks are showing up as orange-highlighted, use the command `/schematicaReplace leaves leaves[decayable=false]` and that will solve the problem.

As per https://github.com/Lunatrius/Schematica/issues/110#issuecomment-267581208 this bug is already known, along with a workaround. I'm merely creating this bug report to keep the issue on the to-do list, as well as assist any future players who are in need of assistance (as I was twenty minutes ago).

The problem boils down to leaf blocks not being saved in/loaded from the .schematic correctly. The leaves are saved as if they grew naturally (block state decayable=true) as opposed to player-placed (block state decayable=false). This causes leaves to be highlighted in orange or destroyed automatically (if that function is enabled) when placed. This has the cute little side effect of making the material list useless for making sure you didn't miss any blocks when it comes to leaves, since it doesn't recognize leaves as being placed correctly.

I have no idea if this is a problem originating with the .schematic format, the mod itself, or how the blocks were placed before being saved (WorldEdit or by hand, for example), but I lean towards the third possibility - I have not done enough testing to be certain.

A solution from the mod's end could be to have Schematica automatically load leaf blocks as decayable=false because the blocks will, under nearly all circumstances, be placed by the player when using Schematica.

However until the bug is fixed, there is a workaround. Simply load your schematic in, and enter the command `/schematicaReplace leaves leaves[decayable=false]`

**Comment by Lunatrius** [comment]
Strictly speaking, this is not a bug. When a schematic is saved, the blocks are taken as they are in the world, without any modifications. That means that loading the schematic back into the same area should **not** highlight any of the blocks unless blocks were changed.

Leaves (and probably some similar blocks) don't work properly with the current system as many have noticed, but there's no way around it without adding a hard coded "fix", which I'm not too keen on. The mod is currently in a rather messy state and I'm surprised that it actually still works for a majority of people.

There's a few reasons why I'm not going to add a hard coded fix for this. One of the reasons is that the code gets (very) hard to maintain. There's all sorts of weird "placeholder fixes" all around the codebase and if parts of that code break it's usually annoying to fix.

*If it isn't broken, don't fix/touch it.* By adding the fix now it's highly unlikely that I'll ever again touch that part of the code. I mean, it *would kind of work*, so why would I bother fixing it again, right?

**Comment by TheGurw** [comment]
That's perfectly reasonable. Not a bug, then. Feature request? A command/setting/etc somewhere to save blockstates as player-placed on initial save?

**Comment by Lunatrius** [comment]
Can be done, but it's again a bit of a problem, because I have to explicitly add support for the block states that are the "player placed" version of the block state.

The changes of `/schematicaReplace` can be made permanent if needed - by just saving the currently loaded/modified schematic. Support for this was added in one of the more recent versions. Just make sure markers are turned off.

**Comment by Moepius** [comment]
For me this workaround doesn't work. I am building custom trees in creative mode per hand, save them as schematic and rebuild them in survival mode on a server. When i'm using the command, the leaves will still be higlighted in orange and when i'm relogging the server, about a half of the leave blocks weren't even placed correctly (ghost blocks).

**Comment by Lunatrius** [comment]
This is not the same issue. The ghost block issue is related to the printer and is usually caused by high ping or an overloaded server. Increasing the delay between block placements should avoid most of the ghost blocks.
